### PR TITLE
[Translator] Deprecate azure-ai-documenttranslator

### PR DESCRIPTION
**Pull request by mssfang**
this package should be deprecated in favor of `azure-ai-translation-document`

### CODE REVIEW

File: .vscode/cspell.json [hunk 1]
```diff
@@ -203,7 +203,6 @@
     "sdk/tools/azure-sdk-archetype/**",
     "sdk/tools/azure-sdk-build-tool/**",
     "sdk/storage/azure-storage-file-share/**",
-    "sdk/translation/azure-ai-documenttranslator/**",
```
Comment (srnagar) [inline .vscode/cspell.json:206]: We should follow the guidelines on deprecating a beta package that has no GA release.

https://eng.ms/docs/products/azure-developer-experience/develop/sdk-deprecation-guidelines

cc: @sandeep-sen

File: sdk/translation/azure-ai-documenttranslator/README.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Azure Document Translator client library for Java
 
+## Deprecation Notice
+
+> **Note:** This SDK is deprecated and will be removed on March 2026. Please plan to migrate to use `azure-ai-translation-document`.
```
Comment (sandeep-sen) [inline sdk/translation/azure-ai-documenttranslator/README.md:5]:
```suggestion
> [!WARNING] 
> This SDK is deprecated and will be removed on March 2026. Please plan to migrate to use `azure-ai-translation-document`.
```

Comment (sandeep-sen) [reply sdk/translation/azure-ai-documenttranslator/README.md:5]: The docs for md alert highlighting
https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

**Review by sandeep-sen — APPROVED** [review]

File: sdk/translation/azure-ai-documenttranslator/README.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Azure Document Translator client library for Java
 
+## Deprecation Notice
```
Comment (srnagar) [inline sdk/translation/azure-ai-documenttranslator/README.md:3]: We also need to update the description in the pom.xml and changelog and release the deprecated package. See [deprecation guidelines for Java](https://github.com/Azure/azure-sdk-for-java/blob/main/doc/deprecation_process.md)

File: sdk/translation/azure-ai-documenttranslator/CHANGELOG.md [hunk 4]
```diff
@@ -1,7 +1,10 @@
 # Release History
 
-## 1.0.0-beta.2 (Unreleased)
+## 1.0.0-beta.2 (2024-12-06)
 
+Please note, this package has been deprecated and will no longer be maintained after May 2026. We encourage you to
```
Comment (srnagar) [inline sdk/translation/azure-ai-documenttranslator/CHANGELOG.md:5]: This is a beta package and will not get any support when marked as deprecated. The source code will be available in the repo as per the deprecation guidelines.
https://eng.ms/docs/products/azure-developer-experience/develop/sdk-deprecation-guidelines

**Review by srnagar — APPROVED** [review]

**Pull request merged**